### DISCUSSION TITLE
v2.14.0

### DISCUSSION THREAD

**Discussion by mikecao** [umami-software/umami#3062] [General]
### Revenue report

We added a new report that allows you to track revenue. Simply send a custom event with revenue data and it will start tracking. [Learn more](https://umami.is/docs/reports/report-revenue).

![image](https://github.com/user-attachments/assets/51fb6167-8e6f-4638-ac3d-87c20396a47c)

### Improved dashboard

In the dashboard, you are now able to select which websites you want to appear, not just the order. Simply toggle the ones you want visible and the rest will be hidden. If none are toggled then all the websites will appear just like before. We also fixed an issue where you couldn't see all your websites if you had too many. Additionally you can now search for websites on the dashboard edit screen.

![image](https://github.com/user-attachments/assets/4dbee4b3-1529-40ce-adb0-aa5c2976f982)

### Hash based routing support

We now record hashes in URLs for websites that use hashed based routing.

### Redirect to team

Umami now redirects to your last selected team on login.

## Fixes
- Fixed dashboard editing #3048
- Fixed unknown countries #3054
- Fixed mobile display issue #3039 
- Fixed wrong events count #2980 
- Fixed display of event properties #2928 
- Fixed session data not deleting #2915

## Updates
- Language updates (German, French, Russian, Czech, German - Switzerland)
- Added minor version tags to Docker images
- Allow underscores in subdomains
- Added cache control headers for tracker script

## Thank you
@quiple @konstantintutsch @Quadrubo @xuserz @peter-kuhmann @yArna @Brawl345 @fra-iesus @ZerGo0 @rafaelschranz @DDDDD12138 @ruchernchong @SigmazZ @michaelpayne02 @chelsey-g @mooxl @eremannisto 

<hr /><em>This discussion was created from the release <a href='https://github.com/umami-software/umami/releases/tag/v2.14.0'>v2.14.0</a>.</em>

**Comment by mulder999** [comment]
Thank you for this new release!

Regarding the hash recording feature, is there an option to disable it either on a per-page basis or globally? Additionally, could there be an opportunity to enhance the UI to allow grouping or management at the page level (ignoring or not the hash part) for better flexibility?

**Comment by tomato-ga** [comment]
I got an error when deploying the latest version of vercel

```
Run yarn install
yarn install v1.22.22
[1/4] Resolving packages...
[2/4] Fetching packages...
warning chart.js@4.4.6: The engine "pnpm" appears to be invalid.
error @typescript-eslint/eslint-plugin@8.15.0: The engine "node" is incompatible with this module. Expected version "^18.18.0 || ^20.9.0 || >=21.1.0". Got "18.1[7](https://github.com/tomato-ga/umami-vercel/actions/runs/12003318805/job/33456605750#step:5:8).1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Process completed with exit code 1.

Run yarn install
yarn install v1.22.22
[1/4] Resolving packages...
[2/4] Fetching packages...
warning chart.js@4.4.6: The engine "pnpm" appears to be invalid.
error @typescript-eslint/eslint-plugin@8.15.0: The engine "node" is incompatible with this module. Expected version "^18.18.0 || ^20.9.0 || >=21.1.0". Got "18.1[7](https://github.com/tomato-ga/umami-vercel/actions/runs/12003318805/job/33456605750#step:5:8).1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Error: Process completed with exit code 1.
```

**Reply by DDDDD12138** [reply]
Your Node version is too old; you need to update it.

**Comment by dukeseb** [comment]
Is there a way to set it up so that the revenue wont be tracked until after a user returns to a thank you page?
